### PR TITLE
Fix various bugs in data loader benchmarking script

### PR DESCRIPTION
**Pull request by adamjstewart**
This PR makes the following changes:

- [x] default stride is now half of default patch size
- [x] remove ROI hacks now that test dataset only contains tiles within CONUS
- [x] `size` and `stride` args of samplers should be in CRS coords, not pixel coords
- [x] default `num_workers` is now 8, provides better performance
- [x] use the same number of data points per epoch in every experiment
- [x] need to concatenate instead of stack image channels

### CODE REVIEW

File: experiments/run_benchmarks_experiments.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
             "--cdl-root",
             CDL_DATA_ROOT,
             "--num-workers",
-            "6",
+            "8",
```
Comment (adamjstewart) [inline experiments/run_benchmarks_experiments.py:40]: My VM has 12 cores, but 8 workers provides the best performance since our batch sizes are powers of 2.

File: torchgeo/datasets/geo.py [hunk 2]
```diff
@@ -286,7 +286,7 @@ def __getitem__(self, query: BoundingBox) -> Dict[str, Any]:
                     filepath = glob.glob(os.path.join(directory, filename))[0]
                     band_filepaths.append(filepath)
                 data_list.append(self._merge_files(band_filepaths, query))
-            data = torch.stack(data_list)
+            data = torch.cat(data_list)
```
Comment (adamjstewart) [inline torchgeo/datasets/geo.py:289]: Not sure exactly how robust this is, but when `rasterio` reads a single Landsat band, the output dimensions are (1, H, W) instead of (H, W) so we need to concatenate along the first axis instead of stacking along a new axis. For files like NAIP where multiple bands are in the same file, the output dimensions are (B, H, W) as expected.

**Review by calebrob6 — APPROVED** [review]
